julia/kernels/bb_template.cxx

```diff
@@ -240,7 +237,7 @@ cudaEvent_t cuda{{{kernel_name}}}::execute(cudaPipelineState& /*pipestate*/, con
         {{^hasbuffer}}
             {{^isoutput}}
                 CHECK_CUDA_ERROR(cudaMemcpyAsync({{{name}}}_memory,
-                                                 {{{name}}}_host.at(gpu_frame_index).data(),
+                                                 {{{name}}}_host.data(),
```

Comment:
Does this copy the complete ringbuffer to the GPU?

This code is for debugging only: It initializes output buffers with a poison value to detect cases where the kernel doesn't properly fill in all its outputs. A simple solution would be to just not copy anything to the GPU any more. Otherwise we should only copy those parts of the buffer that the kernel is allowed to write.

Reply:
Sorry, I was confused. This code is for input buffers, not output buffers, and it is for input buffers that are hidden from Kotekan. I don't think the baseband kernel has any such buffers. The FRB kernel does have such a buffer, and it is only used for the dish layout which is constant in time and which should be handled better anyway – it should be copied to the GPU only once, not for every call. I think your code here is inconsequential in the sense that it is never executed.

Reply:
Okay, thanks for checking that.  I have been meaning to figure out the updatable-config or REST calls, and kotekan protocols for handling constant or infrequently-changed input data.

This specific change was because of a change a while back (#1133/#1150) where we create a set of cudaCommand objects for each step in the GPU pipelines, and each cudaCommand is only responsible for handling one frame at a time - so there is rarely a reason for the cudaCommands to hold GPU-buffer-depth arrays.